src/store/reader.rs

```diff
@@ -33,23 +34,29 @@ pub struct StoreReader {
 
 /// The cache for decompressed blocks.
 struct BlockCache {
-    cache: Mutex<LruCache<usize, Block>>,
-    cache_hits: Arc<AtomicUsize>,
-    cache_misses: Arc<AtomicUsize>,
+    cache: Option<Mutex<LruCache<usize, Block>>>,
```

Comment:
I believe this option is just for being able to use `NonZeroUsize`. I think it's simpler to stick to `cache: Mutex<LruCache<usize, Block>>`. If we want to forbid cache size of zero, an assert can be appropriate.

Reply:
I actually started by propagating the use `NonZeroUsize` through to the code base, but there are places where, indirectly, zero-size caches are created. To continue to support this, I chose the `Option<..>` layer which is not so much about using `NonZeroUsize` but more about support zero-sized caches which `lru` does just not support any more.

Personally, I see two viable options:
* Using `Option<..>` to continue supporting `NonZeroUsize`.
* Propagate the `usize -> NonZeroUsize` type change throughout the API and change any callers to use at least cache size one.

Just asserting a non-zero cache might lead to hard to diagnose panics IMHO as there are multiple layers of indirection between constructing readers, writers, etc. and the usage of `LruCache`.

Reply:
Ok, to keep the change in scope let's stick with Option<...> then.

Reply:
I have the second alternative ready in [a branch](https://github.com/adamreichold/tantivy/tree/disallow-zero-size-block-cache). So if you want that, I can propose that as a follow-up PR since I do agree that having the local change first is much easier to review.